src/pages/admin/profile.tsx

```diff
@@ -0,0 +1,158 @@
+import { VerticalForm, VerticalFormButton, VerticalFormTitle, Cropper } from '@/components/common';
+import { showToast, config } from '@/lib';
+import withAccessType, { GetServerSidePropsWithAuth } from '@/lib/hoc/withAccessType';
+import { PermissionService } from '@/lib/services';
+import { PrivateProfile } from '@/lib/types/apiResponses';
+import { reportError } from '@/lib/utils';
+import { KlefkiAPI } from '@/lib/api';
+
+import type { NextPage } from 'next';
+import Image from 'next/image';
+import { SubmitHandler, useForm } from 'react-hook-form';
+
+import { ChangeEvent, useEffect, useState, useRef } from 'react';
+
+interface UploadBoardPhotoProps {
+  user: PrivateProfile;
+}
+interface FormValues {
+  photo: File;
+}
+const UpdateProfilePage: NextPage<UploadBoardPhotoProps> = ({
+  user: { firstName, lastName },
+}: UploadBoardPhotoProps) => {
+  const { handleSubmit } = useForm<FormValues>();
+  const [file, setFile] = useState<Blob | null>(null);
+  const [croppedImg, setCroppedImg] = useState<Blob | null>(null);
+  const [uploadedURL, setUploadedURL] = useState<string | null>(null);
+  const fileInputRef = useRef<HTMLInputElement>(null); // Reference to the hidden file input
+  const [previewURL, setPreviewURL] = useState<string>('');
+
+  const handleUpload = (e: ChangeEvent<HTMLInputElement>) => {
+    const uploadedFile = e.target.files?.[0] || null;
+    const maxFileSize = 5 * 1024 * 1024;
+
+    if (uploadedFile) {
+      if (!uploadedFile.type.startsWith('image/')) {
+        showToast('Unsupported file type. Please upload an image.');
+        return;
+      }
+      if (uploadedFile.size > maxFileSize) {
+        showToast('File size exceeds the 5MB limit. Please upload a smaller image.');
+        return;
+      }
+      setUploadedURL(null);
+      setFile(uploadedFile);
+    }
+  };
+
+  const handleCrop = async (croppedFile: Blob) => {
+    setCroppedImg(croppedFile);
+
+    setPreviewURL(URL.createObjectURL(croppedFile));
+  };
+
+  const triggerFileInput = () => {
+    if (fileInputRef.current) {
+      fileInputRef.current.click(); // Simulate a click on the hidden file input
+    }
+  };
+
+  const onSubmit: SubmitHandler<FormValues> = async () => {
+    if (!croppedImg) {
+      showToast('Please crop your image to submit!');
+      return;
+    }
+    try {
+      const croppedFile = new File([croppedImg], `${firstName}_${lastName}`, {
+        type: 'image/*',
+      });
```

Comment:
I don't think this is a valid MIME type. If you want to change the file name, you should keep the type of the original file.

```suggestion
      const croppedFile = new File([croppedImg], `${firstName}_${lastName}`, {
        type: croppedImg.type,
      });
```

I think the file name should also include the file extension, like `` `${firstName}_${lastName}.${croppedImg.type.split('/')[1]}` `` or something